### PR TITLE
feat(@desktop/wallet): adding wallet derivation path updates

### PR DESCRIPTION
**Pull request by saledjenic**
Check box added to warn user if they want to add a path out of the default Status' wallet derivation path. In case they do that, they are not able to migrate such keypair to a Keycard.

Corresponding `status-go` PR:
- https://github.com/status-im/status-go/pull/3149

Once we get the final design, we will updated add account modal accordingly.

Closes: #9118

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/panels/DerivationPathsPanel.qml [hunk 1]
```diff
@@ -13,9 +14,19 @@ ColumnLayout {
     id: derivationPathSelect
 
     property string path: ""
+    property bool useFullyCustomPath: true
 
     function reset() {
-        derivationPathInput.text = _internal.defaultDerivationPath
+        console.warn("RESET derivationPathSelect.useFullyCustomPath: ", derivationPathSelect.useFullyCustomPath)
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Wallet/panels/DerivationPathsPanel.qml:20]: leftover?

File: ui/app/AppLayouts/Wallet/panels/DerivationPathsPanel.qml [hunk 2]
```diff
@@ -13,9 +14,19 @@ ColumnLayout {
...
+        if (derivationPathSelect.useFullyCustomPath) {
+            derivationPathFullyCustomInput.text = _internal.customDerivationRootPath
+        }
+        else {
+            derivationPathStatusDefaultInput.text = "1"
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Wallet/panels/DerivationPathsPanel.qml:25]: is this a default value can we make it a readonly property?

File: ui/app/AppLayouts/Wallet/panels/DerivationPathsPanel.qml [hunk 3]
```diff
@@ -55,31 +101,34 @@ ColumnLayout {
...
-            StatusLoadingIndicator {
-                color: Theme.palette.directColor4
+    }
+    StatusInput {
+        id: derivationPathStatusDefaultInput
+        Layout.preferredHeight: 64
+        Layout.preferredWidth: parent.width
+        visible: !derivationPathSelect.useFullyCustomPath
+        maximumHeight: 64
+        text: "1"
```
Comment (Khushboo-dev-cpp) [inline ui/app/AppLayouts/Wallet/panels/DerivationPathsPanel.qml:120]: same as above

**Review by Khushboo-dev-cpp — COMMENTED** [review]
LGTM just some small comments

**Comment by saledjenic** [comment]
@Khushboo-dev-cpp thanks for the comments, all addressed.

**Review by Khushboo-dev-cpp — APPROVED** [review]

**Review by glitchminer — COMMENTED** [review]
### QA

Hi @saledjenic, here are the results of the last test. 

---
1. OPEN CI build crashes when the add account button is clicked but OK from local. 

---
2. OPEN Default path is not `m/44'/60'/0'/0/0` per issue description. Instead it is showing `m/44'/60'/0'/0/1`
image

Notes:
OK: Start of path is greyed out if checkbox is unticked
OK: Path is fully editable if checkbox is ticked
OK: Accounts could be added
OK: If keypair has an account outside the default derivation path then it cannot be migrated
OK: If keypair has an account within the default derivation path then it can be migrated

The exact flow, appearance and language used was not covered as they are subject to new designs that are in progress.

**Comment by saledjenic** [comment]
@glitchminer thanks for the review.

> OPEN CI build crashes when the add account button is clicked but OK from local.

That's known it seems, logged yesterday: https://github.com/status-im/status-desktop/issues/9412
That's not anything connected to the changes I made here, it happens in master as well. It's most likely due to Qt version cause we switched from Qt5.14.2 to Qt5.15.2 and CI works with the latter one and most likely you're using the old one in local.

> OPEN Default path is not m/44'/60'/0'/0/0 per issue description. Instead it is showing m/44'/60'/0'/0/1

Ahhh sorry about the confusion. This is the correct root wallet derivation path `m/44'/60'/0'/0`, but this `m/44'/60'/0'/0/0` is the path of the default wallet account (one displayed in UI as "Status account", one you cannot delete) it's on index `0`. So if user has unchecked checkbox he will be able to change only the last index of the root wallet derivation path, so accounts he may add in that case are like `m/44'/60'/0'/0/1`, `m/44'/60'/0'/0/2`... I updated the description in #9118 


Basically none of above are issues.

**Comment by glitchminer** [comment]
Thanks for the info @saledjenic. I've marked those as closed.

As discussed, this next issue was found where adding an account with default derivition path prevents it from being migrated:

https://user-images.githubusercontent.com/50769329/216587050-0d9e6de3-477f-498b-81a2-be1685f42c7b.mp4

image

**Review by glitchminer — APPROVED** [review]
### QA

Thanks for the fixes @saledjenic. Tested, approved.
cc: @alaibe 

#### Summary

1. OK: Able to migrate when only default account
2. OK: Can add account using default path
3. OK: Add account button only enabled after clicking preview account
4. OK: Checkbox is disabled by default
5. OK: Can migrate after adding using the default derivation path (radio button present)
6. OK: When adding next account using default path the add account remains disabled as index is already taken
7. OK: After changing index +1 then the add account button becomes enabled again
8. OK: Previewed account address matched actual address (except for case)
9. OK: Full path became editable after check box selected. Add account enabled without preview.
10. OK: After adding non-default path account then the keypair no longer has the radio button, a warning is shown when attempting to migrate.
11. OK: After removing account from wallet the account can be migrated again
12. OK: Possible to import seed on non-default derivation path - migration option removed
13. OK: Possible to import seed on default derivation path - migration option present
14. OK: Adding accounts after migration with default and non-default path

Some items noted that could be addressed during redesign:

- The preview only showed the address in lowercase characters only, not the exact address shown in wallet.
- When adding an account after checkbox is enabled the derivation path had an added /0 that was not shown during account creation. For example, user entered `m/44'/60’/0’/1/1/` became `m/44'/60'/0'/1/1/0` in the db.
- Having to authenticate for preview using keycard pin after every amendment to path
